src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs

```diff
@@ -322,18 +325,71 @@ public async Task<ICollection<StatusLookupResult>> LookupPendingTransactionsByIn
         return pendingTransactions.Select(pt => new StatusLookupResult(pt.PayloadHash.ToHex(), pt.Status.ToGatewayModel(), pt.LastFailureReason)).ToArray();
     }
 
-    private async Task<List<GatewayModel.CommittedTransactionInfo>> GetTransactions(List<long> transactionStateVersions, GatewayModel.TransactionCommittedDetailsOptIns optIns, CancellationToken token)
+    private async Task<List<GatewayModel.CommittedTransactionInfo>> GetTransactions(
+        List<long> transactionStateVersions, GatewayModel.TransactionCommittedDetailsOptIns optIns, CancellationToken token)
     {
         var transactions = await _dbContext.LedgerTransactions
             .Where(ult => transactionStateVersions.Contains(ult.StateVersion))
             .ToListAsync(token);
 
         var entityIdToAddressMap = await GetEntityAddresses(transactions.SelectMany(x => x.AffectedGlobalEntities).ToList(), token);
 
-        return transactions
-            .OrderBy(lt => transactionStateVersions.IndexOf(lt.StateVersion))
-            .Select(lt => lt.ToGatewayModel(optIns, entityIdToAddressMap))
+        var schemaHashes = transactions
+            .Where(x => x.EngineReceipt.EventsSchemaHash != null)
+            .SelectMany(x => x.EngineReceipt.EventsSchemaHash!)
             .ToList();
+
+        List<PackageSchemaHistory>? schemas = null;
+
+        if (optIns.ReceiptEvents && schemaHashes.Any())
+        {
+            schemas = await _dbContext.PackageSchemaHistory
+                .Where(x => schemaHashes.Contains(x.SchemaHash))
+                .ToListAsync(token);
+        }
+
+        List<GatewayModel.CommittedTransactionInfo> mappedTransactions = new List<GatewayModel.CommittedTransactionInfo>();
+        var networkId = _networkConfigurationProvider.GetNetworkId();
+
+        foreach (var transaction in transactions.OrderBy(lt => transactionStateVersions.IndexOf(lt.StateVersion)))
```

Comment:
Why do we even order this collection here?

Reply:
we were already doing that before to make sure we return transactions in proper order.

 ```
       return transactions
            .OrderBy(lt => transactionStateVersions.IndexOf(lt.StateVersion))
            .Select(lt => lt.ToGatewayModel(optIns, entityIdToAddressMap))
```